src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1112,53 +1112,26 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
             // RMW intrinsic operands doesn't have to be delayFree when they can be assigned the same register as op1Reg
             // (i.e. a register that corresponds to read-modify-write operand) and one of them is the last use.
 
-            bool op2DelayFree = isRMW;
-            bool op3DelayFree = isRMW;
-            bool op4DelayFree = isRMW;
-
             assert(intrin.op1 != nullptr);
 
-            if (isRMW && intrin.op1->OperIs(GT_LCL_VAR))
+            bool forceOp2DelayFree = false;
+            if ((intrin.id == NI_Vector64_GetElement) || (intrin.id == NI_Vector128_GetElement))
             {
-                unsigned int varNum1    = intrin.op1->AsLclVar()->GetLclNum();
-                bool         op1LastUse = false;
-
-                unsigned int varNum2 = BAD_VAR_NUM;
-                unsigned int varNum3 = BAD_VAR_NUM;
-                unsigned int varNum4 = BAD_VAR_NUM;
 
-                if (intrin.op2->OperIs(GT_LCL_VAR))
+#ifdef DEBUG
+                if (isRMW && intrin.op1->OperIs(GT_LCL_VAR))
                 {
-                    varNum2 = intrin.op2->AsLclVar()->GetLclNum();
-                    op1LastUse |= ((varNum1 == varNum2) && intrin.op2->HasLastUse());
-                }
+                    unsigned int varNum1    = intrin.op1->AsLclVar()->GetLclNum();
+                    bool         op1LastUse = false;
 
-                if (intrin.op3 != nullptr)
-                {
-                    if (intrin.op3->OperIs(GT_LCL_VAR))
+                    unsigned int varNum2 = BAD_VAR_NUM;
+                    if (intrin.op2->OperIs(GT_LCL_VAR))
                     {
-                        varNum3 = intrin.op3->AsLclVar()->GetLclNum();
-                        op1LastUse |= ((varNum1 == varNum3) && intrin.op3->HasLastUse());
+                        varNum2 = intrin.op2->AsLclVar()->GetLclNum();
+                        assert((varNum1 == varNum2) && intrin.op2->HasLastUse());
```

Comment:
The whole block under `ifdef DEBUG` doesn't make sense to me - why are we asserting that for `Vector64.GetElement(vector, index)` and `Vector128.GetElement(vector, index)` their operands correspond the same local? This will never be true - since `op1` is a SIMD value and `op2` is a `int`. 

In fact, I don't think it will ever execute - neither `NI_Vector64_GetElement` nor `NI_Vector128_GetElement` is a RMW intrinsic.

Reply:
That's right. I misread the code thinking that only way `assert(!op2DelayFree);` in previous code would get triggered if `(isRMW == true) && (varNum1 == varNum2)` but didn't realize that it can be because `isRMW == false`. I will remove the `#ifdef DEBUG` block.